### ISSUE TITLE
DocumentContentSynchronizer.documentClosed() returns a useless value

### ISSUE THREAD

**Issue by basilevs** [eclipse/lsp4e#689]
`org.eclipse.lsp4e.DocumentContentSynchronizer.documentClosed()` returns a dummy future that is ignored in LSP4E code.

Given that the main package is not an API, the signature can be safely modified to represent the nature of the method.

**Comment by rubenporras** [comment]
Hi @basilevs , is it not returned by LanguageServerWrapper#disconnet?

**Comment by basilevs** [comment]
> Hi @basilevs , is it not returned by `LanguageServerWrapper#disconnect`?

Yes, but that return value is always ignored.

**Comment by rubenporras** [comment]
Yes, in LSP4E code base, but it can be used. We do indeed use it when manually disconnecting documents

**Comment by basilevs** [comment]
@rubenporras how do you use it? The return value is always a completed Future, so surely, there is nothing useful that can be done with it? (This is idle question, I understand implications of changing signature).

On a more serious note, LSP4E does not declare this package as an API. Do you have related compiler warnings enabled?

**Comment by rubenporras** [comment]
You can chain some operation on the future using `theRun`, `thenApply`, etc. so that you are sure that the code is run only after the disconnect has really happened.
